lib/buffer.js

```diff
@@ -20,6 +20,8 @@ function createPool() {
   poolOffset = 0;
 }
 
+// v8::String::kMaxLength defined in v8.h
+const kStringMaxLength = (1 << 28) - 16;
```

Comment:
I'm inclined to expose this via `require('buffer').kStringMaxLength`. Will make testing easier, and only have a single location to update in case the value changes in the future.